docs/src/remapping.md

```diff
@@ -0,0 +1,159 @@
+interpolation will occur. These arrays are of `Geometry.Point`s.
+
+`ClimaCore.Remapping.interpolate` allocates new output arrays. As such, it is
+not suitable for performance-critical applications.
+`ClimaCore.Remapping.interpolate!` performs interpolation in-place. When using
+the in-place version`, the `dest`ination has to be the same array type as the
+device in use (e.g., `CuArray` for CUDA runs). For performance-critical
+applications, it is preferable to a `ClimaCore.Remapping.Remapper` and use it
+directly (see next Section).
+
+#### Example
+
+Given `field`, a `Field` defined on a cubed sphere.
+
+By default, a target uniform grid is chosen (with resolution `hresolution` and
+`vresolution`), so remapping is
+```julia
+interpolated_array = interpolate(field, hcoords, zcoords)
+```
+Coordinates can be specified:
+
+```julia
+longpts = range(-180.0, 180.0, 21)
+latpts = range(-80.0, 80.0, 21)
+zpts = range(0.0, 1000.0, 21)
+
+hcoords = [Geometry.LatLongPoint(lat, long) for long in longpts, lat in latpts]
+zcoords = [Geometry.ZPoint(z) for z in zpts]
+
+interpolated_array = interpolate(field, hcoords, zcoords)
+```
+The output is defined on the Cartesian product of `hcoords` with `zcoords`.
+
+If the default target coordinates are being used, it is possible to broadcast
+`ClimaCore.Geometry.components` to extract them as a vector of tuples (and then
+broadcast `getindex` to extract the respective coordinates as vectors).
+
+### The `Remapper` object
+
+A `Remapping.Remapper` is an object that is tied to a specified `Space` and can
+interpolate scalar `Field`s defined on that space onto a predefined target grid.
+The grid does not have to be regular, but it has to be defined as a Cartesian
+product between some horizontal and vertical coordinates (meaning, for each
+horizontal point, there is a fixed column of vertical coordinates).
+
+Let us create our first remapper, assuming we have `space` defined on the surface of the sphere
+```julia
+import ClimaCore.Geometry: LatLongPoint, ZPoint
+import ClimaCore.Remapping: Remapper
+
+hcoords = [Geometry.LatLongPoint(lat, long) for long in -180.:180., lat in -90.:90.]
+remapper = Remapper(space, target_hcoords)
+```
+This `remapper` object knows can interpolate `Field`s defined on `space` with the same `interpolate` and `interpolate!` functions.
+```julia
+import ClimaCore.Fields: coordinate_field
+import ClimaCore.Remapping: interpolate, interpolate!
+
+example_field = coordinate_field(space)
+interpolated_array = interpolate(remapper, example_field)
+
+# Interpolate in place
+interpolate!(interpolated_array, remapper, example_field)
+```
+
+!!! note
+
+    `interpolate` allocates new arrays and has some internal
+     type-instability, `interpolate!` is non-allocating and type-stable.
+     When using `interpolate!`, the destination has to be defined on the
+     root process and has to be `nothing` for the other MPI processes.
```

Comment:
Can we remove this note and mention the MPI-related piece above, where we mention the destination array type requirements?